daemon/cmd/cni.go

```diff
@@ -0,0 +1,331 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright Authors of Cilium
+
+package cmd
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"os"
+	"path"
+	"strings"
+	"text/template"
+	"time"
+
+	"github.com/cilium/cilium/pkg/controller"
+	"github.com/cilium/cilium/pkg/option"
+
+	"github.com/google/renameio"
+	"github.com/tidwall/sjson"
+)
+
+// Some functions around managing the CNI configuration file
+//
+// By default, we write the CNI configuration file to disk after the
+// daemon has successfully started up.
+
+// cniConfigs are the default configurations, per chaining mode
+var cniConfigs map[string]string = map[string]string{
+	// the default
+	"none": `
+{
+  "cniVersion": "0.3.1",
+  "name": "cilium",
+  "plugins": [
+    {
+       "type": "cilium-cni",
+       "enable-debug": {{.Debug | js }},
+       "log-file": "{{.LogFile | js }}"
+    }
+  ]
+}`,
+
+	"flannel": `
+{
+  "cniVersion": "0.3.1",
+  "name": "flannel",
+  "plugins": [
+    {
+      "type": "flannel",
+      "delegate": {
+         "hairpinMode": true,
+         "isDefaultGateway": true
+      }
+    },
+    {
+      "type": "portmap",
+      "capabilities": {
+        "portMappings": true
+      }
+    },
+    {
+       "type": "cilium-cni",
+       "enable-debug": {{.Debug | js }},
+       "log-file": "{{.LogFile | js }}"
+    }
+  ]
+}
+`,
+	"portmap": `
+{
+  "cniVersion": "0.3.1",
+  "name": "portmap",
+  "plugins": [
+    {
+       "type": "cilium-cni",
+       "enable-debug": {{.Debug | js }},
+       "log-file": "{{.LogFile | js }}"
+    },
+    {
+      "type": "portmap",
+      "capabilities": {"portMappings": true}
+    }
+  ]
+}
+`,
+}
+
+// The CNI plugin config we inject in to the plugins[] array of existing AWS configs
+const awsCNIEntry = `
+{
+	"type": "cilium-cni", 
+	"enable-debug": {{.Debug | js }}, 
```

Comment:
Ultra-nit: trailing whitespaces
```suggestion
	"type": "cilium-cni",
	"enable-debug": {{.Debug | js }},
```

Reply:
Hah, I fixed this, but then my laptop died and took the hard drive with it :-)